sorc/chgres_cube.fd/model_grid.F90

```diff
@@ -781,12 +795,331 @@ subroutine define_input_grid_gfs_grib2(localpet, npets)
+ real(esmf_kind_r4), allocatable       :: longitude_one_tile(:,:), lon_corners(:,:)
+ real(esmf_kind_r8)                    :: lat_target(i_target,j_target), &
+                                          lon_target(i_target,j_target)
+ real(esmf_kind_r8)                    :: deltalon, dx
+ integer                               :: ncid,id_var, id_dim
+ real(esmf_kind_r8), pointer           :: lat_src_ptr(:,:), lon_src_ptr(:,:)
+ character(len=10000)            :: cmdline_msg, temp_msg, temp_msg2
+ character(len=10)              :: temp_num = 'NA'
+
+ num_tiles_input_grid = 1
+
+ !inv_file = "chgres.inv"
+ the_file = trim(data_dir_input_grid) // "/" // grib2_file_input_grid
+ temp_file = trim(fix_dir_input_grid)//"/latlon_grid3.32769.nc"
+
+ call ESMF_FieldGather(latitude_target_grid, lat_target, rootPet=0, tile=1, rc=error)
+ if(ESMF_logFoundError(rcToCheck=error,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+      call error_handler("IN FieldGather", error)
+ call ESMF_FieldGather(longitude_target_grid, lon_target, rootPet=0, tile=1, rc=error)
+ if(ESMF_logFoundError(rcToCheck=error,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+      call error_handler("IN FieldGather", error)
+
+
+  ! Check for the grid template number to see if wgrib2 can access the lat/lon arrays
+  if (localpet==0) then
+    cmdline_msg = "wgrib2 "//trim(the_file)//" -d 1 -Sec3 &> temp.out"
+    call system(cmdline_msg)
+  endif
+  call MPI_BARRIER(MPI_COMM_WORLD, error)
+  open(4,file="temp.out")
+  read(4,"(A)") temp_msg
+  close(4)
+  i = index(temp_msg, "Template=") + len("Template=")
+  
+  j = index(temp_msg," opt arg")
+  temp_num=temp_msg(i:j-1)
+
+  ! Wgrib2 can't properly read the lat/lon arrays of data on NCEP rotated lat/lon
+  ! grids, so read in lat/lon from fixed coordinate file
+  print*, 'temp num =', temp_num
+  if (trim(temp_num)=="3.32769") then
+
+     input_grid_type = "rotated_latlon"
+
+     error=nf90_open(trim(temp_file),nf90_nowrite,ncid)
+     call netcdf_err(error, 'opening: '//trim(temp_file))
+
+     error=nf90_inq_dimid(ncid, 'nx', id_dim)
+     call netcdf_err(error, 'reading nx id' )
+     error=nf90_inquire_dimension(ncid,id_dim,len=i_input)
+     call netcdf_err(error, 'reading nx value' )
+
+     error=nf90_inq_dimid(ncid, 'ny', id_dim)
+     call netcdf_err(error, 'reading ny id' )
+     error=nf90_inquire_dimension(ncid,id_dim,len=j_input)
+     call netcdf_err(error, 'reading ny value' )
+
+     allocate(longitude_one_tile(i_input,j_input))
+     allocate(latitude_one_tile(i_input,j_input))
+
+     error=nf90_inq_varid(ncid, 'gridlat', id_var)
+     call netcdf_err(error, 'reading field id' )
+     error=nf90_get_var(ncid, id_var, latitude_one_tile)
+     call netcdf_err(error, 'reading field' )
+
+     error=nf90_inq_varid(ncid, 'gridlon', id_var)
+     call netcdf_err(error, 'reading field id' )
+     error=nf90_get_var(ncid, id_var, longitude_one_tile)
+     call netcdf_err(error, 'reading field' )
+     
```

Comment:
Is this instance of grb2_mk_inv being called on all tasks?  This is instance and the one at line 897 be moved before the 'if' statement?

Reply:
Thanks, @GeorgeGayno-NOAA.  This instance is now behind an appropriate if statement to run as a single task.  I just pushed the commit.

Reply:
@GeorgeGayno-NOAA, the print statement was also moved into the "if" statement too.

Reply:
Sorry, I meant to ask if the instances at 886 and 899 could be combined and moved before the 'if' at 856.

Reply:
If we have it create an inventory file before checking the template number, it's possible the inventory file could be created for a template number that won't be compatible with the code, and then it would fail. I didn't think that was very efficient.

Reply:
Ok.  That answers my question.